app/views/dashboard/firms/_form.html.erb

```diff
@@ -0,0 +1,59 @@
+<% if @firm.errors.any? %>
+  <div class="t-validation-summary">
+    <%= f.validation_summary %>
+  </div>
+<% end %>
+
+<section class="form-section">
+  <div class="form-section__heading-with-button">
+    <% if f.object.trading_name? %>
+      <h2 class="form-section__heading-with-button__heading"><%= t('dashboard.firm_edit.trading_name_heading') %></h2>
+    <% else %>
+      <h2 class="form-section__heading-with-button__heading"><%= t('dashboard.firm_edit.firm_heading') %></h2>
+      <a class="form-section__heading-with-button__button button button--primary"><%= t('dashboard.add_trading_name_button') %></a>
```

Comment:
Nesting is allowed?! Dang, why didn't ya tell me. This changes everything. Will implement as you suggest.

I can't remember why the button stuff was there — I think it was a remnant from trying to override something that was coming from a few places. Anyway, we clearly didn't need it in the end!